src/test/java/org/prebid/server/bidder/taboola/TaboolaBidderTest.java

```diff
@@ -0,0 +1,400 @@
+
+        // when
+        final Result<List<BidderBid>> result = taboolaBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .extracting(BidderError::getMessage)
+                .containsOnly("Failed to find banner/native impression \"123\"");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsReturnEmptyListsResultWhenEmptySeatBidInBidResponse() throws JsonProcessingException {
+        // given
+        final BidRequest bidRequest = BidRequest.builder().build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                bidRequest,
+                mapper.writeValueAsString(BidResponse.builder().build()));
+
+        // when
+        final Result<List<BidderBid>> result = taboolaBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnValidBidResponseWithBannerTypeWhenImpIncludeNativeAndBanner()
+            throws JsonProcessingException {
+        // given
+        final BidRequest bidRequest = givenBidRequest(bidRequestCustomizer -> bidRequestCustomizer,
+                givenImp(impCustomizer -> impCustomizer
+                        .id("123")
+                        .xNative(Native.builder().build())
+                        .banner(Banner.builder().build())));
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                bidRequest,
+                mapper.writeValueAsString(
+                        givenBidResponse(identity())));
+
+        // when
+        final Result<List<BidderBid>> result = taboolaBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.banner);
+    }
+
+    @Test
+    public void makeBidsShouldReturnValidBidResponseWithBannerWhenRequestHaveTwoImp() throws JsonProcessingException {
+        // given
+        final Imp impBanner = givenImp(impCustomizer ->
+                        impCustomizer.id("123").banner(Banner.builder().build()),
+                extImpCustomizer -> extImpCustomizer
+                        .publisherId("publisher_video_Id")
+                        .bidFloor(BigDecimal.TEN)
+                        .publisherDomain("test.com")
+                        .tagId("tag_video"));
+        final Imp impNative = givenImp(impCustomizer ->
+                        impCustomizer.id("124").xNative(Native.builder().build()),
+                extImpCustomizer -> extImpCustomizer
+                        .publisherId("publisher_native_Id")
+                        .bidFloor(BigDecimal.TEN)
+                        .publisherDomain("test.com")
+                        .tagId("tag_native"));
```

Comment:
This imp (id=124) is redundant. It does zero affect on test (none of the bids corresponds to this imp).
Please, remove it.